cmd/snap/cmd_run.go

```diff
@@ -118,6 +118,37 @@ func getSnapInfo(snapName string, revision snap.Revision) (*snap.Info, error) {
 	return info, nil
 }
 
+func createOrUpdateUserDataSymlink(info *snap.Info, usr *user.User) error {
+	// 'current' symlink for user data (SNAP_USER_DATA)
+	userData := info.UserDataDir(usr.HomeDir)
+	wantedSymlinkValue := filepath.Base(userData)
+	currentActiveSymlink := filepath.Join(userData, "..", "current")
+
+	var err error
+	var currentSymlinkValue string
+	for i := 0; i < 5; i++ {
+		currentSymlinkValue, err = os.Readlink(currentActiveSymlink)
+		// failure other than non-existing symlink is fatal
+		if err != nil && !os.IsNotExist(err) {
+			// TRANSLATORS: %v the error message
+			return fmt.Errorf(i18n.G("cannot read symlink: %v"), err)
+		}
+		if currentSymlinkValue == wantedSymlinkValue {
+			break
+		}
+		if !os.IsNotExist(err) {
+			_ = os.Remove(currentActiveSymlink)
```

Comment:
Can you please check for the kind of error that is returned from `os.Remove`? Errors other than ENOENT should not be ignored. As for ENOENT itself let's please add a comment similar to this one:

```
// We may be racing with other instances of snap-run that try to do the same thing
// If the symlink is already removed then we can ignore this error and try again.
```

Then we should do a `continue`, perhaps coupled with a tiny sleep value to ensure we get rescheduled.